Firestore/core/src/firebase/firestore/remote/grpc_streaming_reader.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_REMOTE_GRPC_STREAMING_READER_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_REMOTE_GRPC_STREAMING_READER_H_
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "Firestore/core/src/firebase/firestore/remote/grpc_completion.h"
+#include "Firestore/core/src/firebase/firestore/util/async_queue.h"
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+#include "grpcpp/client_context.h"
+#include "grpcpp/generic/generic_stub.h"
+#include "grpcpp/support/byte_buffer.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+/**
+ * Sends a single request to the server, reads one or more streaming server
+ * responses, and invokes the given callback with the accumulated responses.
+ */
+class GrpcStreamingReader {
```

Comment:
Two obvious alternatives would be:
* don't create a new wrapper class, use gRPC objects directly in `Datastore`;
* implement `GrpcStreamingReader` in terms of `GrpcStream`.

For point 1, I think this class is large enough to justify its existence. 2 seems a little convoluted, though I'd like to hear other perspectives as well.